build_runner/test/package_graph/package_graph_test.dart

```diff
@@ -217,6 +217,96 @@ void main() {
     });
   });
 
+  group('orderedPackageCycles', () {
+    test('with two sub trees', () {
+      var a = new PackageNode('a', '1.0.0', PackageDependencyType.path, null);
+      var left1 =
+          new PackageNode('left1', '1.0.0', PackageDependencyType.pub, null);
+      var left2 =
+          new PackageNode('left2', '1.0.0', PackageDependencyType.pub, null);
+      var right1 =
+          new PackageNode('right1', '1.0.0', PackageDependencyType.pub, null);
+      var right2 =
+          new PackageNode('right2', '1.0.0', PackageDependencyType.pub, null);
+      a.dependencies.addAll([left1, right1]);
+      left1.dependencies.add(left2);
+      right1.dependencies.add(right2);
+      var graph = new PackageGraph.fromRoot(a);
+      var inOrder = graph.orderedPackageCycles.map((c) => c.map((p) => p.name));
+      expect(
```

Comment:
nit: for all of these tests add an expect for the total length of `inOrder`.

Reply:
Done